bme280/bme280.go

```diff
@@ -52,9 +69,34 @@ func New(bus drivers.I2C) Device {
 	}
 }
 
-// Configure sets up the device for communication and
-// read the calibration coefficientes.
+// ConfigureWithSettings sets up the device for communication and
+// read the calibration coefficients.
+//
+// The default configuration is the Indoor Navigation settings
+// from the BME280 datasheet.
 func (d *Device) Configure() {
+	d.ConfigureWithSettings(Config{})
+}
+
+// ConfigureWithSettings sets up the device for communication and
+// read the calibration coefficients.
+//
+// The default configuration if config is left at defaults is
+// the Indoor Navigation settings from the BME280 datasheet.
+func (d *Device) ConfigureWithSettings(config Config) {
+	d.Config = config
+
+	// If config is not initialized, use Indoor Navigation defaults.
+	if d.Config == (Config{}) {
```

Comment:
What happens if the user only set some of the values of the configuration? They will default to 0. I think it's better to check every parameter, like other drivers do.

Also, is there any need for d.Config to be exported?

Reply:
I'm not sure I follow, comparing against the 'default' struct is the same as comparing all fields?

Since the default values in the struct are valid values for the fields, I don't think there's anything more to do?

Reply:
Maybe I'm not understanding it right, this is the configuration if left empty, right?:

```go
Config{
			Mode:        ModeNormal,
			Period:      Period0_5ms,
			Temperature: Sampling2X,
			Humidity:    Sampling1X,
			Pressure:    Sampling16X,
			IIR:         Coeff16,
		}
```

Now, If I just want to set one parameter:

```go
sensor.ConfigureWithSettings(bme280.Config{Humidity: bme280.Sampling2X})
```

I get the following configuration (if I'm not wrong):

```go
d.Config = Config{
			Mode:        ModeSleep, // 0
			Period:      Period0_5ms, // 0 
			Temperature: SamplingOff, // 0
			Humidity:    Sampling2X, // value set by the user
			Pressure:    SamplingOff, // 0
			IIR:         Coeff0, // 0
		}
```

What I expected/wanted: 

```go
Config{
			Mode:        ModeNormal, 
			Period:      Period0_5ms,
			Temperature: Sampling2X,
			Humidity:    Sampling2X, // value set by the user
			Pressure:    Sampling16X,
			IIR:         Coeff16,
		}
```


For me it's a bit strange behavior, I'm forced to set all the parameters instead of only the ones that differs from the "default/none" configuration. Or maybe I'm missing something.

Reply:
Oh, I see - the problem i hit is that `SamplingOff` (Zero) is a valid value, if Configure logic was...
```
if config.Pressure == SamplingOff {
   d.config.Pressure = Sampling2X
}
```

There would be no way to disable sampling of pressure - setting it to 'off' would always be changed to 2x.

One alternative could be to make all the parameters pointers and check against nil?   But that would make it tedious to initialize the config structure?

The other alternative is to make zero an invalid value by bumping all the consts by 1 - which complicates the driver logic in that `0` in the device register would now be `1` in the config struct.  That would also make the register constants less descriptive and couldn't be re-used by app code reading the registers directly.

I'm open for other ways to do defaults...

Reply:
Ok, it's not the behavior I'll expected but I guess it's good enough.

I only could think of defining doing something like 
```go
const (
	SamplingOff Oversampling = iota+1
	Sampling1X
	Sampling2X
	Sampling4X
	Sampling8X
	Sampling16X
)
```

But then you'll have to do : 
```go
d.Config.Temperature = config.Temperature-1
d.Config.Mode = mode-1
...
```

Maybe a bit too much, so better to leave it as it is? What do you think @deadprogram

Reply:
FYI - looking at periph.io, they've got for a different model for 'default values' (see https://github.com/google/periph/commit/0e5d546d9ab1f630fc2a1e43cdfb06e7b7b8665d)

I don't think we should adopt just for one driver - but it seems a cleaner model in some ways, and could help solve the problem in machine package where default I2C pins create a back-dependency between boards and chipsets.

Reply:
Also created a proposal to adopt something similar to the periph.io model across TinyGo: https://github.com/tinygo-org/tinygo/issues/3152